applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -74,6 +74,9 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
         if (GetGeometry().LocalSpaceDimension() == 1) {
             GetGeometry().DeterminantOfJacobian(det_J_container, this->GetIntegrationMethod());
             dN_dX_container = GetGeometry().ShapeFunctionsLocalGradients(this->GetIntegrationMethod());
+            for (unsigned int index = 0; index < dN_dX_container.size(); ++index) {
+                dN_dX_container[index] /= det_J_container[index];
+            }
```

Comment:
Just learnt, there is w = element_div(u,v) which does a division of the each element of u vector by correspondent element of v vector. Would it work for you?